go/vt/vttablet/tabletmanager/vreplication/vcopier.go

```diff
@@ -244,13 +244,13 @@ func (vc *vcopier) copyTable(ctx context.Context, tableName string, copyState ma
 			}
 			fieldEvent := &binlogdatapb.FieldEvent{
 				TableName: initialPlan.SendRule.Match,
-				Fields:    rows.Fields,
 			}
+			fieldEvent.Fields = append(fieldEvent.Fields, rows.Fields...)
```

Comment:
I'm not following. isn't `fieldEvent.Fields` guaranteed to be nil since we just created the struct? Why append to the field instead of just setting it?

Reply:
This is an idiomatic (maybe not that idiomatic, since you haven't heard about it! :sweat_smile:) way of copying `rows.Fields`. Since `fieldEvent.Fields` is always `nil`, but `nil` is a valid slice, this is doing the same thing as:

```go
fieldEvent.Fields = make([]pb.Field, len(rows.Fields)
copy(fieldEvent.Fields, rows.Fields)
```

...Which is something we need to do since now we're using memory pooling. We cannot set it directly to `rows.Fields` or that slice will change when we return the rows to the pool.

Reply:
That's what I guessed it was for, but I was uncertain.

If it said `fieldEvent.Fields = append(nil, rows.Fields...)` instead I would probably have not reacted, but here it looked weird.